ui/src/components/mobileservices/BindingPanel.js

```diff
@@ -205,7 +196,42 @@ export class BindingPanel extends Component {
     return errors;
   };
 
+  filterPlatforms() {
+    if (this.props.service.isUPSService()) {
+      const hasIOS = this.hasUPSIOSBoundService();
+      const hasAndroid = this.hasUPSAndroidBoundService();
+      const { schema } = this.state;
+      if (!hasAndroid && !hasIOS) {
+        if (schema.properties.CLIENT_TYPE) {
+          schema.properties.CLIENT_TYPE.default = 'Android';
+          schema.properties.CLIENT_TYPE.enum = ['Android', 'IOS'];
+        }
+      } else if (hasAndroid && !hasIOS) {
+        // UPS, there's already an Android variant
+        if (schema.properties.CLIENT_TYPE) {
+          schema.properties.CLIENT_TYPE.default = 'IOS';
+          schema.properties.CLIENT_TYPE.enum = ['IOS'];
+        }
+      } else if (!hasAndroid && hasIOS) {
+        // UPS, there's already an IOS variant
+        if (schema.properties.CLIENT_TYPE) {
+          schema.properties.CLIENT_TYPE.default = 'Android';
+          schema.properties.CLIENT_TYPE.enum = ['Android'];
+        }
+      } else if (hasAndroid && hasIOS) {
+        // UPS, there's already an IOS variant
+        if (schema.properties.CLIENT_TYPE) {
+          schema.properties.CLIENT_TYPE.default = '';
+          schema.properties.CLIENT_TYPE.enum = [];
+        }
+      }
+      // we don't care if there are variants for both platforms.
+      // this binding panel shouldn't be shown anyway
```

Comment:
do we need the `else if (hasAndroid && hasIOS)` case then?

Reply:
It won't be needed anymore when AEROGEAR-8273 will be resolved